docs/enhancements/image-service-handle-huge-images.md

```
@@ -0,0 +1,110 @@
+
+When an HTTP server supports the Accept-Ranges: bytes header (which almost all static file servers and CDNs do), a client can request a specific byte chunk by sending a header like Range: bytes=1000-2000.
+To avoid downloading the ISO, a custom Go struct has to be implemented that satisfies the io.ReadSeeker interface but translates those calls into HTTP requests:
+* Seek(offset, whence): Updates an internal integer tracking the "current position." No network call is made yet.
+* Read(p []byte): Translates the internal position and the size of p into an HTTP Range request, executes the GET, and writes the response body into p.
+Assisted-image-service uses diskfs for costumizing the ISO9660 filesystem. To find the offset of /images/ignition.img or other files diskfs must read the ISO header, jump over volume descriptor, jump to the root directory and traverse the filesystem tree. This requires multiple read and seek operations.
+Once the boundaries are found the image can be seemlesly streamed, with the overlay injecting the custom data as the stream passes those byte offsets.
+To overcome the constant exploration of the ISO filesystem (discovering the boundaries), the offsets of the required files can be cashed (the ISO is immutable, meaning that the data and it's location does not change).
+When an offline image is requested the flow looks like:
+* one http stream is opened
+* in a loop
+** reads the data from the http server into a small buffer
```

markdownlint-cli2 (0.23.2)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)